op-node/rollup/derive/l1_block_info.go

```diff
@@ -200,15 +227,28 @@ func (info *L1BlockInfo) marshalBinaryEcotone() ([]byte, error) {
 	return w.Bytes(), nil
 }
 
+func (info *L1BlockInfo) unmarshalBinaryIsthmus(data []byte) error {
+	return unmarshalBinaryEcotoneAndIsthmus(info, data, true)
+}
 func (info *L1BlockInfo) unmarshalBinaryEcotone(data []byte) error {
+	return unmarshalBinaryEcotoneAndIsthmus(info, data)
+}
+
+func unmarshalBinaryEcotoneAndIsthmus(info *L1BlockInfo, data []byte, isIsthmus ...bool) error {
```

Comment:
I think its best to not have a function `ecotoneAndIsthmus` and instead have 2 functions. I would prefer a copy/paste of code in this case, its consensus critical code and we should optimize for easy to read. If we want to deduplicate code, we could create a function that accepts the 4byte signature + into + data and call that inside of both the ecotone and isthmus functions, hardcoding the appropriate 4byte sig

Reply:
got it. I'll do:
> we could create a function that accepts the 4byte signature + into + data and call that inside of both the ecotone and isthmus functions, hardcoding the appropriate 4byte sig

it's quite a big function, and I rather have the "same" code, since as you said, is quite sensitive, as to avoid any mishaps.